Tests/unicodedata/scripts_test.py

```diff
@@ -0,0 +1,179 @@
+from __future__ import (
+    print_function, division, absolute_import, unicode_literals)
+from fontTools.misc.py23 import *
+
+
+from fontTools import unicodedata
+
+
+def test__binary_search_range():
+    ranges = [
+        (0, 10, "foo"),
+        (11, 11, "bar"),
+        (12, 20, "baz"),
+        (21, 22, "bum"),
+    ]
+    search = unicodedata._binary_search_range
+
+    for i in range(11):
+        assert search(1, ranges) == "foo"
+
+    assert search(11, ranges) == "bar"
+
+    for i in range(12, 21):
+        assert search(i, ranges) == "baz"
+
+    for i in range(21, 23):
+        assert search(i, ranges) == "bum"
+
+    assert search(23, ranges) is None
+
+
+def test_script():
+    assert unicodedata.script("a") == "Latin"
+
+    assert hasattr(unicodedata.script, "cache")
+    assert unicodedata.script("a") == unicodedata.script.cache["a"]
+
+    # these were randomly sampled, one character per script
```

Comment:
I would add some Unknowns, too, right before and after an existing script range.

Reply:
Indeed, I thought the ranges were continuous. Aren’t they? I’ll check

Reply:
Hm, it seems that they are, sorry!

Reply:
Actually, I checked wrong. Look here:

```
0376..0377    ; Greek # L&   [2] GREEK CAPITAL LETTER PAMPHYLIAN DIGAMMA..GREEK SMALL LETTER PAMPHYLIAN DIGAMMA
037A          ; Greek # Lm       GREEK YPOGEGRAMMENI
```
There's nothing between `0377` and `037A`.

Reply:
Ah! I’m away from the laptop so I can’t check. I wonder what my script function returns in between those. It should return “Unknown”. If it doesn’t it’s a bug

Reply:
But the ranges in the text file are not sorted by codepoint, so it might be that those codepoints are covered by some other script.
In theory all codepoints should be assigned a script...

Reply:
Check the generated scripts.py, same gap.

Reply:
There should be tons of gaps between codepoints within a script range because the block is reserved for that script, but not all codepoints are used yet, if ever. I tried some I know like Chakma 0x11145 which is proposed and almost accepted, but not officially added yet. So right now it returns Unknown as expected.

Reply:
So it works. Thanks for checking.